src/esm_runscripts/oasis.py

```diff
@@ -431,73 +455,89 @@ def add_restart_files(self, restart_file, fconfig):
-                                f"``{restart_file}``"
-                            )
-                    else:
-                        if not gconfig["isinteractive"]:
-                            # If more than one restart file found that matches ini_restart_date,
-                            # ask the user to select from the result list:
-                            message = (
-                                "More than one OASIS restart file was found for "
-                                "your branchoff experiment that matches the "
-                                "ini_restart_date you selected. Please select "
-                                "one of the following OASIS restart files:"
-                            )
-                            answers = questionary.form(
-                                restarts = questionary.select(message, choices=glob_restart_file)
-                            ).ask()
-                            restart_file = os.path.basename(answers["restarts"])
-
-                config["restart_in_sources"][restart_file_label] = restart_file
-
-        if restart_file not in config["restart_in_sources"]:
+            else:
+                glob_search_file = restart_file_path
+
+            glob_restart_file = glob.glob(glob_search_file)
+            glob_restart_file.sort()
+            if restart_file and is_runtime:
+                # If there are more than one file found let the user decide which one to take
+                if len(glob_restart_file) == 1:
+                    restart_file = os.path.basename(glob_restart_file[0])
+                elif len(glob_restart_file) == 0:
+                    restart_file = restart_file_path
+                    if not os.path.isfile(restart_file):
+                        user_error(
+                            "Restart file missing",
+                            f"No OASIS restart file for ``{restart_file_label}`` found "
+                            f"matching the pattern ``{glob_search_file}`` nor "
+                            f"``{restart_file}``"
+                        )
+                else:
+                    if not gconfig["isinteractive"]:
+                        # If more than one restart file found that matches ini_restart_date,
+                        # ask the user to select from the result list:
+                        message = (
+                            "More than one OASIS restart file was found for "
+                            "your branchoff experiment that matches the "
+                            "ini_restart_date you selected. Please select "
+                            "one of the following OASIS restart files:"
+                        )
+                        answers = questionary.form(
+                            restarts = questionary.select(message, choices=glob_restart_file)
+                        ).ask()
+                        restart_file = answers["restarts"]
```

Comment:
I like this a lot 😄

Reply:
@nwieters work, not just one indentation level up, for a more generalized case